### PR TITLE
Fix minor test issues

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Address several minor test issues.

# Details

These were noticed when working on #1414. They look like a combination of cut-n-paste confusion and general-confusion-with-the-arguments-to-Data2DInt

### CODE REVIEW

File: sherpa/tests/test_data.py
Comment (hamogu) [inline sherpa/tests/test_data.py:720]: I note that these two tests can now we summarized as:
```
@pytest.mark.parametrize("data", Data2DClasses, indirect=True)
def test_data2_get_x1(data):
    numpy.testing.assert_array_equal(data.get_x1(), X1_2D)
```
That's not important (it only saves three lines) so I'm approving, but if @DougBurke wants to adjust that before merging thats also fine.

**Review by hamogu — APPROVED** [review]

File: sherpa/tests/test_data.py
Comment (DougBurke) [reply sherpa/tests/test_data.py:720]: There's a number of other tests that would benefit from this cleanup, but I don't have the time/energy to do this so I'd rather not do it here.

**Pull request merged**
